React/Tailwind/Best practices.md

```diff
@@ -0,0 +1,272 @@
+
+### Can Tailwind coexist with libraries like Material UI?
+
+Yes. Tailwind is just CSS classes; it does not add a runtime. Two integration patterns work well:
+
+- Simply use the utility classes in MUI components
+  ```tsx
+  <Button className="bg-brand-600 shadow-md" />
+  ```
+- Use Tailwind for layout, MUI for logic - Tailwind controls flex grids, spacing, and colors, while the MUI theme handles stateful styles (disabled, focus)
+
+Important: make sure `tailwind.css` is imported after any baseline or component library stylesheet so its utilities win specificity ties.
+
+### How do I handle dark mode?
+
+Use Tailwind's `dark:` variant together with CSS custom properties:
+
+```css
+@theme {
+  --brand-500: 22 119 255;
+}
+
+.dark {
+  --brand-500: 147 197 253;
+}
+```
+
+Then in JSX:
+
+```tsx
+className = "bg-brand-500 dark:bg-brand-500/80";
+```
+
+Switching the dark class on `<html>` node flips every token instantly without component re-renders.
+
+### What is the `@utility` directive and when should I reach for it?
+
+`@utility` lets you bundle several raw CSS declarations under a single Tailwind-style class without leaving your stylesheet. It is perfect for verbose properties such as `clip-path` or complex shadows that would otherwise require an arbitrary value every time.
+
+```css
+@utility {
+  .clip-blob {
+    clip-path: polygon(50% 0%, 100% 25%, 100% 75%, 50% 100%, 0% 75%, 0% 25%);
+  }
+}
+```
+
+You can now reuse `clip-blob` alongside built-in utilities with complete purge safety and no JavaScript.
+
+### How does the group utility help with state-based styling?
+
+Instead of repeating the same prefix on multiple siblings (`hover:`, `focus:`), wrap them in a parent that carries `group`:
+
+```tsx
+<button className="group inline-flex items-center gap-2 p-3 rounded-lg bg-brand-600">
+  <Icon className="size-4 text-white group-hover:translate-x-0.5 transition-transform" />
+  <span className="text-white group-hover:underline">Download</span>
+</button>
+```
+
+Any child can now react to `group-hover`, `group-focus`, `group-aria-expanded`, etc., keeping class strings short and readable.
+
+### My hover and focus class lists are gigantic. Is there a cleaner pattern?
+
+Use CVA or the `cn` helper to build a map of state variants instead of stacking prefixes manually:
+
+```tsx
+const card = cva("rounded-lg p-4 transition", {
+  variants: {
+    intent: {
+      default: "",
+      hover: "hover:bg-brand-50 hover:shadow",
+      focus: "focus-visible:ring-2 focus-visible:ring-brand-600",
+    },
+  },
+});
+```
+
+Apply the variant you need with `card({ intent: 'hover' })`. Literal strings remain visible for the compiler and you avoid prefix noise.
+
+### How do I configure Material UI so Tailwind utilities take effect?
```

Comment:
Maybe a small chapter on how to have a custom value in class? As a n00b I would like to know that information. 

```
bg-[#ccc]
p-[13px]
```